source/routing/raptor_api.cpp

```diff
@@ -185,20 +185,46 @@ static void co2_emission_aggregator(pbnavitia::Journey* pb_journey){
     }
 }
 
-static void compute_durations(pbnavitia::Journey* pb_journey) {
+
+static void compute_metadata(pbnavitia::Journey* pb_journey) {
     uint32_t total_walking_duration = 0;
+    uint32_t total_car_duration = 0;
+    uint32_t total_bike_duration = 0;
+    uint32_t total_walking_distance = 0;
+    uint32_t total_car_distance = 0;
+    uint32_t total_bike_distance = 0;
+
     for (const auto& section: pb_journey->sections()) {
-        if ((section.type() == pbnavitia::STREET_NETWORK && section.street_network().mode() == pbnavitia::StreetNetworkMode::Walking) ||
-        (section.type() == pbnavitia::TRANSFER && section.transfer_type() == pbnavitia::walking) ||
-        (section.type() == pbnavitia::CROW_FLY && section.street_network().mode() == pbnavitia::StreetNetworkMode::Walking)) {
+        if (section.type() == pbnavitia::STREET_NETWORK || section.type() == pbnavitia::CROW_FLY) {
+            if (section.street_network().mode() == pbnavitia::StreetNetworkMode::Walking) {
+                total_walking_duration += section.duration();
+                total_walking_distance += section.length();
+            } else if (section.street_network().mode() == pbnavitia::StreetNetworkMode::Car) {
+                total_car_duration += section.duration();
+                total_car_distance += section.length();
+            } else if ((section.street_network().mode() == pbnavitia::StreetNetworkMode::Bike) ||
+                       (section.street_network().mode() == pbnavitia::StreetNetworkMode::Bss)) {
+                total_bike_duration += section.duration();
+                total_bike_distance += section.length();
+            }
+
+        } else if (section.type() == pbnavitia::TRANSFER && section.transfer_type() == pbnavitia::walking) {
```

Comment:
Minor : maybe also test that transfer_type exists (if None, then it's walking)

Reply:
It can be a stay_in, thus the test is needed